src/ChannelServer/Skills/Cleric/Heal.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Melia.Channel.Network;
+using Melia.Channel.Skills.Base;
+using Melia.Channel.World.Entities;
+using Melia.Shared.Const;
+using Melia.Shared.Util;
+using Melia.Shared.World;
+
+namespace Melia.Channel.Skills.Cleric
+{
+	/// <summary>
+	/// Handler for the Archer skill Multishot.
+	/// </summary>
+	[SkillHandler(SkillId.Cleric_Heal)]
+	class Heal : IGroundSkillHandler
+	{
+		/// <summary>
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="castPosition"></param>
+		/// <param name="targetPosition"></param>
+		public void Handle(Skill skill, Character caster, Position castPosition, Position targetPosition)
+		{
+			// This is just a test for a dedicated skill handler, the code
+			// was largely copied from the default handler for now, without
+			// referencing logs. And the packets are definitely not right
+			// yet.
+
+			var castRange = castPosition.Get3DDistance(targetPosition);
+			if (castRange > skill.Data.MaxRange)
+			{
+				Log.Warning("Thrust: Player {0} cast skill farther than max range ({1} > {2}).", caster.Name, castRange, skill.Data.MaxRange);
+				return;
+			}
+
+			if (skill.SpendSp > 0)
+				caster.ModifySp(-skill.SpendSp);
+
+			skill.IncreaseOverheat();
+
+			Send.ZC_SKILL_READY(caster, skill, caster.Position, targetPosition);
+
+			// The hitbox seems pretty small, there's presumably more going
```

Comment:
So splash range isn't the correct value here altogether, from what I figure, that value is the skill after cast "splashing" to the next monster. While we're treating it as the target range of the initial cast. For skills that are held to cast and draw a green range on the floor use the values from skill_selectground_drawrange.ies. For example, 30001 Archer Multishot has a draw range of 30. Not too sure for other skills.

Reply:
I thought splash range might be a catch-all. It might mean different things for the different target types (though as I commented, there's surely some kind of calculation involved). For a `Circle` type it might mean the radius, while for a `Square` it might be the width/height, and for Thrust, which uses `Square`, I figured it might be the width of the imagined rectangle that is draw between source and target position, to designate the hit box. I think skill_selectground_drawrange has decidedly too few entries to apply to a lot of skills. And doesn't it sound like it's primarily for _drawing_? So the client know how to display the target area? Your interpretation of splash being solely for splash makes sense, but how is the target area defined then?

Reply:
Not a part of this PR, marking as resolved for now.